packages/compass-aggregations/src/components/stage.tsx

```diff
@@ -0,0 +1,162 @@
+
+const DragHandleToolbar = sortableHandle((props: { index: number }) => {
+  return <StageEditorToolbar {...props}></StageEditorToolbar>
+});
+
+const stageStyles = css({
+  position: 'relative',
+  margin: spacing[3],
+  display: 'flex',
+  flexDirection: 'row',
+  alignItems: 'stretch',
+  overflow: 'hidden' // this is so that the top left red border corner does not get cut off when there's a server error
+});
+
+const stageWarningStyles = css({
+  borderColor: palette.yellow.base
+});
+
+const stageErrorStyles = css({
+  borderColor: palette.red.base
+});
+
+const stageEditorNoPreviewStyles = css({
+  width: '100%'
+});
+
+const stagePreviewContainerStyles = css({
+  display: 'flex',
+  position: 'relative',
+  flexDirection: 'column',
+  width: '100%',
+  overflow: 'auto'
+});
+
+const RESIZABLE_DIRECTIONS = {
+  top: false,
+  right: true,
+  bottom: false,
+  left: false,
+  topRight: false,
+  bottomRight: false,
+  bottomLeft: false,
+  topLeft: false
+};
+
+type ResizableEditorProps = {
+  index: number,
+  isExpanded: boolean,
+  isAutoPreviewing: boolean,
+};
+
+function ResizableEditor({ index, isExpanded, isAutoPreviewing }: ResizableEditorProps) {
+  const editor = (
+    <>
+      <DragHandleToolbar index={index} />
+      {isExpanded && (
+        // @ts-expect-error typescript is getting confused about the index prop. Requires stage-editor.jsx to be converted.
+        <StageEditor index={index} />
+      )}
+    </>
+  );
+
+  if (!isAutoPreviewing) {
+    return <div className={stageEditorNoPreviewStyles}>{editor}</div>;
+  }
+
+  return (
+    <Resizable
+      defaultSize={{
+        width: '388px',
+        height: 'auto',
+      }}
+      minWidth="260px"
+      maxWidth="92%"
+      enable={RESIZABLE_DIRECTIONS}
+      handleComponent={{
+        right: <ResizeHandle />,
+      }}
+      handleStyles={{
+        right: {
+          // Position the line in the middle of the container so that:
+          // a) It sits on the border of the editor and preview areas rather
+          //    than inside the editor.
+          // b) The user initiates dragging from the line and not slightly off
+          //    to the left.
+          // If this ever needs to be tweaked, the easiest way is to give the
+          // editor and preview toolbars different background colours and add a
+          // transparent background here.
+          right: '-9px' // default -5px
+        }
+      }}
+    >
+      {editor}
+    </Resizable>
+  );
+}
+
+const DEFAULT_OPACITY = 0.6;
+
+type StageProps = {
+  index: number,
+  isEnabled: boolean,
+  isExpanded: boolean,
+  hasSyntaxError: boolean,
+  hasServerError: boolean,
+  isAutoPreviewing: boolean
+}
+
+function Stage({
+  index,
+  isEnabled,
+  isExpanded,
+  hasSyntaxError,
+  hasServerError,
+  isAutoPreviewing
+}: StageProps) {
+  const opacity = isEnabled ? 1 : DEFAULT_OPACITY;
+
+  return (<KeylineCard className={cx(
+      stageStyles,
+      hasSyntaxError && stageWarningStyles,
```

Comment:
Currently we are not showing any warning if the stage was just added (no operator or value has been set and as a result syntaxError is null).

Reply:
I pushed a commit and now it shouldn't show the yellow border if there is no stage operator or value.

Reply:
(behavior for the banner should be unchanged and it will still show the yellow border if there is some other syntax error. ie. if there is an operator and value)